ambry-replication/src/main/java/com/github/ambry/replication/RemoteReplicaInfo.java

```diff
@@ -198,6 +204,24 @@ public boolean equals(Object obj) {
     return false;
   }
 
+  /**
+   * Gets the cached meta data response information received in the previous replication cycle.
+   * Replication manager calls this method to check if there are any missing store messages that are now obtained via intra-dc replication.
+   * @return exchangeMetadataResponse which contains the meta data response information (missing keys, token info, local lag from remote, etc.).
+   */
+  synchronized ReplicaThread.ExchangeMetadataResponse getExchangeMetadataResponse() {
+    return exchangeMetadataResponse;
+  }
+
+  /**
+   * Stores/caches the meta data response received in the previous replication cycle.
+   * Replica threads calls this method to store the metadata response for standby replicas.
+   * @param exchangeMetadataResponse contains the meta data response information (missing keys, token info, local lag from remote, etc.).
+   */
+  synchronized void setExchangeMetadataResponse(ReplicaThread.ExchangeMetadataResponse exchangeMetadataResponse) {
```

Comment:
I need a little bit context here: when is the case to `setExchangeMetadataResponse `?

Reply:
I am trying to understand why this method should be `synchronized` ?